test/node-rpc-test.js

```diff
@@ -68,11 +70,75 @@ describe('RPC', function() {
     assert.deepEqual(info.localservicenames, ['NETWORK', 'WITNESS']);
   });
 
+  it('should rpc getblockhash', async () => {
+    const info = await nclient.execute('getblockhash', [node.chain.tip.height]);
+    assert.strictEqual(util.revHex(node.chain.tip.hash), info);
+  });
+
+  describe('Blockchain', function () {
+    it('should rpc getchaintips', async () => {
+      const info = await nclient.execute('getchaintips', []);
+      assert.strictEqual(info.length, 1);
+      assert.strictEqual(util.revHex(node.chain.tip.hash), info[0].hash);
+    });
+
+    it('should rpc getchaintips when a block is mined', async () => {
```

Comment:
So - there is a problem with this test, it should not pass! We are adding a blockchain fork and yet the number of chain tips remains at 1, this is incorrect. I finally realized why: it's because this regtest node is totally fresh and only has one block (the genesis block)! If you add `console.log(entry)` after line 86 you will get `null` and so the block you create is just block 1, and does not compete with any other block. Therefore there is still only one chain tip.

If you move this entire `describe('Blockchain')` section down after the `getblock` section you will see this test fail. Why? Because the `getblock` section adds blocks to the chain. So when you create your block at height `node.chain.tip.height - 1` an actual fork is the result, and the # of chaintips actually goes up to 2 ;-)

This is an important lesson: Didn't you expect the number of chaintips to be 2? If you started with that but the test failed, the solution is not to just change the test vector back to `1`, but to ask yourself more deeply why the test failed ;-)

Reply:
I fixed it by creating a chain fork right from the `genesis` block, and we have two chains, one with four blocks and the other with three blocks. Hence there will be two chain tips.